canonical-web-and-design/maas-ui
# refactor(ui): extract common FilesystemFields component from storage forms

## Done

- Created `FilesystemFields` component which houses some common fields found in a few of the storage forms
- Made formik field values consistent between forms (so they're all `fstype` instead of `filesystemType`, which also matches the api)
- Fixed some incorrect behaviour with the filesystem fields:
  - Removed requirement for a mount point if a fstype is selected. This is only meant to be the case for special filesystems, which was the first of the filesystem forms to be built.
  - Changed the fields to disabled instead of hidden if a fstype is not selected. We're trying to move away from hiding elements.
  - Fixed handling of "swap" filesystems

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the storage tab of a Ready or Allocated machine
- Open the "Add partition" form for a disk - you should see the disabled form fields for mount point and mount options
- Check that selecting a filesystem enables the mount point and mount options fields
- Check that you can add a partition with a filesystem even if it doesn't have a mount point
- Unmount the filesystem and edit it
- Change the filesystem to "swap"
- Check that the mount point field is disabled with "none" as the placeholder

## Fixes

Fixes #1970
